### PR TITLE
Handle autorange axes with 'reset' doubleClick config option [fixes #278]

### LINKED ISSUE #278: Exception when doubleClick: "reset" is set
IIf you set the doubleClick option to "reset", an exception is triggered when double-clicking the graph.

Sample:
http://codepen.io/aleburato/pen/adePBZ

### PR DESCRIPTION
**Pull request by etpinard**
@mdtusz 

- A three-line bug fix
- Some linting
- 200+ of tests

### CODE REVIEW

File: src/plots/cartesian/axes.js [hunk 1]
```diff
@@ -202,16 +202,18 @@ axes.saveRangeInitial = function(gd, overwrite) {
     var axList = axes.list(gd, '', true),
         hasOneAxisChanged = false;
 
-    var ax, isNew, hasChanged;
-
     for(var i = 0; i < axList.length; i++) {
-        ax = axList[i];
+        var ax = axList[i];
```
Comment (mdtusz) [inline src/plots/cartesian/axes.js]: I'm always happy to see these kinds of small style changed. No more scrolling all the way to the top of a file to see whether some variable had a value previously!

File: test/jasmine/tests/click_test.js [hunk 2]
```diff
@@ -87,4 +93,232 @@ describe('click interactions', function() {
...
+
+            mockCopy.layout.yaxis.autorange = false;
+            mockCopy.layout.yaxis.range = setRangeY.slice();
+
+            return mockCopy;
+        }
+
+        describe('when set to \'reset+autorange\' (the default)', function() {
+            it('should work when \'autorange\' is on', function(done) {
+                Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(function(){
```
Comment (mdtusz) [inline test/jasmine/tests/click_test.js]: Can we try to flatten these :christmas_tree:'s?

File: test/jasmine/tests/click_test.js [hunk 3]
```diff
@@ -87,4 +93,232 @@ describe('click interactions', function() {
...
+
+        });
+
+        describe('when set to \'reset\'', function() {
+            var config = {
+                doubleClick: 'reset'
+            };
+
+            it('should work when \'autorange\' is on', function(done) {
+                Plotly.plot(gd, mockCopy.data, mockCopy.layout, config).then(function() {
```
Comment (mdtusz) [inline test/jasmine/tests/click_test.js]: Here too.

File: src/plots/cartesian/axes.js (hunk 1 above)
Comment (etpinard) [inline src/plots/cartesian/axes.js]: Yeah. I used to dislike block-scoped `var`, but I've changed.

**Comment by mdtusz** [comment]
Good tests. Just flatten the callback :christmas_tree:'s out, then :dancer:!

File: test/jasmine/tests/click_test.js (hunk 2 above)
Comment (etpinard) [inline test/jasmine/tests/click_test.js]: Good call.

**Pull request merged**